### PR TITLE
Implement Index/MultiIndex.equals

### PR DESCRIPTION
**Pull request by HyukjinKwon**
```python
import databricks.koalas as ks
idx = ks.Index(['a', 'b', 'c'])
midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
idx.equals(idx)
midx.equals(idx)
from databricks.koalas.config import option_context
with option_context('compute.ops_on_diff_frames', True):
    idx.equals(ks.Index(['a', 'b', 'c']))
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I am merging this for today's release.

File: databricks/koalas/indexes.py [hunk 1]
```diff
@@ -225,21 +225,64 @@ def identical(self, other):
...
+        >>> from databricks.koalas.config import option_context
+        >>> idx = ks.Index(['a', 'b', 'c'])
+        >>> idx.name = "name"
+        >>> midx = ks.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
+        >>> midx.names = ("nameA", "nameB")
+
+        For Index
+
+        >>> idx.equals(idx)
+        False
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: Why is this `False`? Also, for `identical`?

```py
>>> import pandas as pd
>>> idx = pd.Index(['a', 'b', 'c'])
>>> idx.equals(idx)
True
>>> idx.identical(idx)
True
```

File: databricks/koalas/indexes.py [hunk 2]
```diff
@@ -225,21 +225,64 @@ def identical(self, other):
...
+        >>> with option_context('compute.ops_on_diff_frames', True):
+        ...     idx.equals(ks.Index(['b', 'b', 'a']))
+        False
+        >>> idx.equals(midx)
+        False
+
+        For MultiIndex
+
+        >>> midx.equals(midx)
+        False
```
Comment (ueshin) [inline databricks/koalas/indexes.py]: ditto.

```py
>>> midx = pd.MultiIndex.from_tuples([('a', 'x'), ('b', 'y'), ('c', 'z')])
>>> midx.names = ("nameA", "nameB")
>>> midx.equals(midx)
True
>>> midx.identical(midx)
True
```

File: databricks/koalas/indexes.py (hunk 1 above)
Comment (ueshin) [reply databricks/koalas/indexes.py]: Submitted a PR #1220.

**Pull request merged**